### PR TITLE
Run Unit Tests in Docker using a GitHub action

### PR DESCRIPTION
**Pull request by ned21**
Setup a GitHub action to run the unit tests in a Docker container on every pull request to master.

I have not yet worked out how to make this Action running successfully a pre-merge status check but hopefully it will be obvious once it is merged.

### CODE REVIEW

**Comment by ned21** [comment]
We have 44 PRs queued up, and AFAICT no way to check the unit tests pass before merging them.  Any chance of someone taking a look at this soon?

File: .github/workflows/continuous-integration.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  push:
+    branches: [ github-action ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
```
Comment (stdweird) [inline .github/workflows/continuous-integration.yaml:13]: does this matter wrt the `centos:7` below?

**Comment by stdweird** [comment]
@ned21 so you want this merged , and then you hope the github actions can be used? 

wrt the unittests, we run following PRs on top of master: 1348,1370,1379,1390,1399,1400,1401,1402,1403,1421,1422,1423,1424,1427,1428,1432,1433,1434,1438,1444

they are build as usual (ie unittests run before rpm is made), so these should all pass

i can do another round of jenkins poking, but what PRs do you want merged first?

**Comment by stdweird** [comment]
@ned21 jenkins tests are triggered again. not sure why they were disabled (there was a 'Active' checkbox unchecked somehow)

File: .github/workflows/continuous-integration.yaml (hunk 1 above)
Comment (ned21) [reply .github/workflows/continuous-integration.yaml:13]: No - the ubuntu machine just runs the Docker image so we end up running centos with a ubuntu kernel. :)

**Comment by ned21** [comment]
@stdweird - cool! I thought Jenkins machine might be dead. Now 19.12 is out, hopefully we can start merging things again if Jenkins gives them the green light?

**Pull request merged**